pyproj/_crs.pyx

```diff
@@ -215,97 +473,439 @@ cdef class Ellipsoid:
+            proj_coordoperation_has_ballpark_transformation(
+                projcontext,
+                coord_operation.projobj
+            )
+
+        # TODO: How do you get the value count?
+        # The example in the PROJ test just says 7 ....
+        # cdef double *out_values
+        # cdef int value_count = ?
+        # cdef int emit_error_if_incompatible = 0
+        # proj_coordoperation_get_towgs84_values(
+        #     projcontext,
+        #     coord_operation.projobj,
+        #     double *out_values,
+        #     int value_count,
+        #     int emit_error_if_incompatible
+        # )
+
+        coord_operation.projctx = get_pyproj_context()
+        return coord_operation
+
+    @property
+    def params(self):
+        """
+        Returns
+        -------
+        list[Param]: The coordinate operation parameters.
+        """
+        if self._params is not None:
+            return self._params
+        self._params = []
+        cdef int num_params = 0
+        num_params = proj_coordoperation_get_param_count(
+            self.projctx,
+            self.projobj
+        )
+        for param_idx from 0 <= param_idx < num_params:
+            self._params.append(
+                Param.create(
+                    self.projctx,
+                    self.projobj,
+                    param_idx
+                )
+            )
+        return self._params
+
+    @property
+    def grids(self):
+        """
+        Returns
+        -------
+        list[Grid]: The coordinate operation grids.
+        """
+        if self._grids is not None:
+            return self._grids
+        self._grids = []
+        cdef int num_grids = 0
+        num_grids = proj_coordoperation_get_grid_used_count(
+            self.projctx,
+            self.projobj
+        )
+        for grid_idx from 0 < grid_idx < num_grids:
+            self._grids.append(
+                Grid.create(
+                    self.projctx,
+                    self.projobj,
+                    grid_idx
+                )
+            )
+        return self._grids
+
+    def to_proj4(self, version=5):
+        """
+        Convert the projection to a PROJ.4 string.
+
+        Parameters
+        ----------
+        version: int
+            The version of the PROJ.4 string. Default is 5.
+
+        Returns
+        -------
+        str: The PROJ.4 string.
+        """
+        return _to_proj4(self.projctx, self.projobj, version)
+
+
+
+cdef class _CRS(Base):
+    """
+    The cython CRS class to be used as the base for the
+    python CRS class.
+    """
+    def __cinit__(self):
+        self._proj_type = PJ_TYPE_UNKNOWN
         self._ellipsoid = None
         self._area_of_use = None
         self._prime_meridian = None
+        self._datum = None
+        self._sub_crs_list = None
+        self._source_crs = None
+        self._coordinate_system = None
+        self._coordinate_operation = None
+
+    def __init__(self, projstring):
```

Comment:
Is this `__init__` actually needed? (I mean the user facing functionality part of it). 
Because now that means that you can do both `CRS("..some string..")` as `CRS.from_string("..some string..")` ? Which means we give users two ways to do exactly the same, I would be fine to only have the `from_..` methods as the user facing way to create a CRS object.

(note: this is not really related to the rest of the PR, so can also open an issue to separate that out)

Reply:
Definitely a thought for a separate issue. I have mixed feelings about it at the moment.

Reply:
> Definitely a thought for a separate issue. I have mixed feelings about it at the moment.

Opened https://github.com/pyproj4/pyproj/issues/271 for this

Reply:
Thanks!